umm-spec-lib/src/cmr/umm_spec/time.clj

```diff
@@ -4,6 +4,21 @@
   (:require
     [clj-time.core :as t]))
 
+(defn temporal-ranges
+  "Returns all the :RangeDateTimes and :SingeDateTimes contained in the given TemporalExtent record,
+   as a list of {:BeginningDateTime xxx :EndingDateTime xxx}"
+  [temporal]
+  (let [ranges (:RangeDateTimes temporal)
+        ;; when :EndsAtPresentFlag is set to true, set :EndingDateTime for each range to nil.
+        ;; otherwise, set :EndingDateTime to nil if it doesn't exist.
+        normalized-ranges (if (:EndsAtPresentFlag temporal) 
+                            (map #(assoc % :EndingDateTime nil) ranges)
+                            (map #(if (= nil (:EndingDateTime %)) (assoc % :EndingDateTime nil) %) ranges))
```

Comment:
Since the result of this will need to map to the nested mapping which contains both the start-date and end-date, I just want to make sure both BeginningDateTime and EndingDateTime are present, even with nil value. I don't know what would happen to the nested mapping if an element is missing.